[Feature] #50 : 트레이닝의 트레이너 정보 조회 API

saym010116 opened the pull request:
## 📑 이슈 번호
- close #50 

## ✨️ 작업 내용
- [x] 트레이너 상세 조회 API 구현 – 트레이너 프로필 및 개설한 트레이닝 목록 조회 기능 추가.
- [x] 인증 여부(isCertified) 처리 – trainer_certificates 존재 여부를 기반으로 인증 여부 반환.

## 💙 코멘트 (선택)

## 📸 구현 결과 (선택)
<img width="621" height="638" alt="image" src="https://github.com/user-attachments/assets/97411ff2-e3c0-471a-adc4-7e2eb0aa8612" />

saym010116 committed:
feat(trainer-detail): 컨트롤러에 트레이너 상세 조회 엔드포인트 추가

saym010116 committed:
feat(trainer-detail): 서비스 레이어에서 트레이너 상세 조회 로직 구현

saym010116 committed:
feat(trainer-detail): 트레이너 상세 조회 Mapper 인터페이스 추가

saym010116 committed:
Feat: 트레이너 상세 조회 SQL 매퍼 XML 작성

saym010116 committed:
Feat: 트레이너 상세 응답 DTO 추가

saym010116 committed:
Feat: 트레이닝 요약 응답 DTO 추가

saym010116 committed:
Feat: 트레이너 프로필 상세 DTO 추가

saym010116 committed:
Fix: 파일 명명 변경

sangeune reviewed (approved):
LGTM

Chaeniiiii commented on src/main/java/com/kbulkup/training/dto/response/TrainerDetailResponseDTO.java:
```diff
@@ -4,15 +4,15 @@
     private List<TrainerTrainingSummaryDTO> trainings;
 
-    public TrainerDetailResponseDTO(TrainingTrainerDetailProfileResonseDTO trainer,
+    public TrainerDetailResponseDTO(TrainingTrainerDetailProfileResponseDTO trainer,
```
**`P1`**
@AllArgsConstructor, @NoArgsConstructor를 추가하시면 생성자가 자동으로 생성되어 해당 코드를 작성할 필요가 없습니다 !

Chaeniiiii commented on src/main/java/com/kbulkup/training/dto/response/TrainerDetailResponseDTO.java:
```diff
@@ -4,15 +4,15 @@
     }
 
-    public TrainingTrainerDetailProfileResonseDTO getTrainer() { return trainer; }
+    public TrainingTrainerDetailProfileResponseDTO getTrainer() { return trainer; }
```
**`P1`**
@Getter를 추가하시면 모든 getter 메서드를 자동 생성하여 해당 코드를 작성할 필요가 없습니다 !

saym010116 committed:
Refactor: 서비스에서 DTO 조립하도록 수정
- TrainerService 인터페이스를 단일 메서드(getTrainerDetail)만 노출
- Impl에서 내부적으로 프로필과 트레이닝 목록 조회 후 DTO 조립

saym010116 committed:
Refactor: DTO에 Lombok 적용
- @Getter, @AllArgsConstructor, @NoArgsConstructor 추가
- 불필요한 수동 생성자 제거

saym010116 committed:
Refactor: 컨트롤러 단순화
- TrainerDetailResponseDTO를 서비스에서 조립하여 반환하도록 수정

HyoSangByun commented on src/main/java/com/kbulkup/training/service/TrainerServiceImpl.java:
```diff
@@ -0,0 +1,24 @@
+    public TrainerDetailResponseDTO getTrainerDetail(Long trainerId) {
+        TrainingTrainerDetailProfileResponseDTO profile = traineeTrainerMapper.selectTrainerProfile(trainerId);
+        List<TrainerTrainingSummaryDTO> trainings = traineeTrainerMapper.selectTrainerTrainings(trainerId);
+        return new TrainerDetailResponseDTO(profile, trainings);
```
``` java
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TrainerDetailResponseDTO {

    private TrainingTrainerDetailProfileResponseDTO trainer;
    private List<TrainerTrainingSummaryDTO> trainings;

    public static TrainerDetailResponseDTO create(TrainingTrainerDetailProfileResponseDTO trainer,
                                              List<TrainerTrainingSummaryDTO> trainings) {
        return TrainerDetailResponseDTO.builder()
                .trainer(trainer)
                .trainings(trainings)
                .build();
    }
}
```
`p1`: 이렇게 해서 팩토리 메서드로 DTO 반환해주세요!

saym010116 committed:
Refactor: 팩토리 메서드를 사용하여 객체 생성하는 방식으로 변경
